bot/exts/events/advent_of_code/_helpers.py

```diff
@@ -151,7 +151,22 @@ def _parse_raw_leaderboard_data(raw_leaderboard_data: dict) -> dict:
         # this data to JSON in order to cache it in Redis.
         daily_stats[day] = {"star_one": star_one, "star_two": star_two}
 
-    return {"daily_stats": daily_stats, "leaderboard": sorted_leaderboard}
+        per_day_star_stats = collections.defaultdict(list)
+        for member in raw_leaderboard_data.values():
+            username = member["name"] if member["name"] else f"Anonymous #{member['id']}"
+
+            for day_data, stars_data in member["completion_day_level"].items():
+                # Iterate over the complete stars for this day for this participant
+                for star, data in stars_data.items():
+                    # Record completion of this star for this individual
+                    completion_time = int(data["get_star_ts"])
+                    per_day_star_stats[f"{day_data}-{star}"].append(
+                        {'completion_time': completion_time, 'member_name': username}
+                    )
```

Comment:
I believe it can be. Initially I wanted to modify _StarResults_ but then decided to go with that and forgot that It could be solved there. Good catch! Thank you!